### ISSUE TITLE
PipeTransport + pickRemoteProcess fails. Returns "Illegal variable name". Wrong quotes escaping.

### ISSUE THREAD

**Issue by pirpyn** [microsoft/vscode-cpptools#6184]
Type: Debugger

**Describe the bug**
- OS and Version: Windows 10 1909 (Local) / RHEL 7.4 (remote-ssh) / RHEL 6.5 (pipeTransport)
- VS Code Version: 1.49.1
- C/C++ Extension Version: 1.0.1
- Other extensions you installed (and if the issue persists after disabling them): Not related

When using the pipeTransport functionality in attach mode, using pickRemoteProcess for processId, the output panel remote-attach returns 

```bash
Illegal variable name.
Command failed: "ssh" "XXXXXX" 'sh -c "uname && if [ $(uname) = \"Linux\" ] ; then ps axww -o pid=,comm=aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa,args= ; elif [ $(uname) = \"Darwin\" ] ; then ps axww -o pid=,comm=aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa,args= -c; fi"'
Illegal variable name.
```

**To Reproduce**
Need no program
```jsonc
{
    "version": "0.2.0",
    "configurations": [
        {
            "name": "attach",
            "type": "cppdbg",
            "request": "attach",
            "miDebuggerPath": "/usr/bin/gdb",
            "program": "/usr/bin/gdb",
            "processId": "${command:pickRemoteProcess}",
            "MIMode": "gdb",
            "pipeTransport": {
                "pipeCwd": "/tmp",
                "pipeProgram": "ssh",
                "pipeArgs": [
                    "XXXXXX",
                ],
                "debuggerPath": "/usr/bin/gdb",
            },
        },
    ]
}
```

**In Console**
Running the reported command in shell fails with the same message.

**Reason** 
Quoting quotes is tedious and rather strange: see [shell pitfalls](https://mywiki.wooledge.org/BashPitfalls#A.5B_.24foo_.3D_.22bar.22_.5D) and [quoting quotes](https://grymoire.com/unix/Quote.html).

TL;DR: The idea is to leave the quote context to quote quotes.  Let's assume you don't have access to double quotes, then
* BAD:  `'Don\'t quote me'.`
* GOOD `'Don'\''t quote me'.`

Also the true sh command substitution is the backquote, not the bash $( ) way, ( which usually work because sh->/bin/bash, except for me I guess, stuck with csh/tcsh. ).
When we want a full sh compliance, we must deal with the outdated backquote.
These character must also be escaped or they will be expanded before the call to ssh, meaning uname will be from the remote-ssh'd machine, not the pipeTransport one.

An easy way to test that is to change the start of the command for 
```
"sh -c 'exit; uname && if [ "\"`unam`\""
```
a message will return that `unam` is unkown, even though we exit before; whereas with 
```
"sh -c 'exit; uname && if [ "\"\`unam\`\""
```
the message will only be issued if we remove the exit command.

**Proposition for a fix**

The command 
```shell
'sh -c "uname && if [ $(uname) = \"Linux\" ] ; then ps axww -o pid=,comm=aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa,args= ; elif [ $(uname) = \"Darwin\" ] ; then ps axww -o pid=,comm=aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa,args= -c; fi"'
```
should be
```shell
$ "sh -c 'uname && if [ "\"\`uname\`\"" = Linux ] ; then ps axww -o pid=,comm=aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa,args= ; elif [ "\"\`uname\`\"" = Darwin ] ; then ps axww -o pid=,comm=aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa,args= -c ; fi'"
```

**Comment by pirpyn** [comment]
The option "C_Cpp.debugger.useBacktickCommandSubstitution": true solves the issue
